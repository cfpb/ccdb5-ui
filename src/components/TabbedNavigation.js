import './TabbedNavigation.less';
import { useDispatch, useSelector } from 'react-redux';
import getIcon from './iconMap';
import React from 'react';
import { selectQueryTab } from '../reducers/query/selectors';
import { changeTab } from '../reducers/query/query';

export const TabbedNavigation = () => {
  const dispatch = useDispatch();
  const tab = useSelector(selectQueryTab);

  const getTabClass = (selectedTab) => {
    const tabName = selectedTab.toLowerCase() + ' tab';
    return tab === selectedTab ? tabName + ' active' : tabName;
  };

  return (
    <div className="tabbed-navigation" data-tour="tabbed-navigation">
      <section>
        <button
          className={getTabClass('Trends')}
          onClick={() => dispatch(changeTab('Trends'))}
        >
          {getIcon('chart')}
          Trends
        </button>

        <button
          className={getTabClass('List')}
          onClick={() => dispatch(changeTab('List'))}
        >
          {getIcon('list')}
          List
        </button>

        <button
          className={getTabClass('Map')}
          onClick={() => dispatch(changeTab('Map'))}
        >
          {getIcon('map')}
          Map
        </button>
      </section>
    </div>
  );
};
