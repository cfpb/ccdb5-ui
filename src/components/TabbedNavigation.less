@import (less) "../css/base.less";

.tabbed-navigation {
  margin-top: 30px;
  border-bottom: 1px solid @gray-20;
  section {
    padding-left: 10px;
    .tab {
      margin: 0;
      padding: 4px 30px 8px 30px;

      font-size: unit( 19px / @base-font-size-px, em );
      font-weight: 500;

      &:focus {
        outline: none;
      }

      .cf-icon-svg {
        margin-right: 5px;
        margin-top: 4px;
        font-size: unit( 13px / @base-font-size-px, em );
      }

      &:not(.active) {
        color: @dark-pacific;
        background: @gray-5;
        border: 1px solid @gray-20;
        border-bottom: none;

        &:after {
          content: '';
          border-top: solid 1px;
          display: block;
        }
      }
      &.active {
        padding-top: 8px;
        border: 1px solid @gray-40;
        background: @white;
        pointer-events: none;
        position: relative;

        &:after {
          content: '';
          display: block;
          width: calc( 100% - 2px);
          outline: 1px solid @white;
          position: absolute;
          bottom: -1px;
          left: 1px;
        }
      }
    }
  }
}
