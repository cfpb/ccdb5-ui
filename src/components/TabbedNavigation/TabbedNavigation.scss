@use 'sass:math';
@use '@cfpb/cfpb-design-system/src/abstracts/index' as *;
@use 'css/base';

.tabbed-navigation {
  margin-top: 30px;
  border-bottom: 1px solid var(--gray-20);
  section {
    padding-left: 10px;
    .tab {
      margin: 0;
      padding: 4px 30px 8px;

      font-size: math.div(19px, $base-font-size-px) + em;
      font-weight: 500;

      .cf-icon-svg {
        margin-right: 5px;
        margin-top: 4px;
        font-size: math.div(13px,  $base-font-size-px) + em;
      }

      &:not(.active) {
        color: var(--pacific-dark);
        background: var(--gray-5);
        border: 1px solid var(--gray-20);
        border-bottom: none;

        &:hover {
          color: var(--navy-dark);
        }

        &:focus {
          color: var(--navy-dark);
        }

        &::after {
          content: '';
          border-top: solid 1px;
          display: block;
        }
      }
      &.active {
        padding-top: 8px;
        border: 1px solid var(--gray-40);
        background: var(--white);
        pointer-events: none;
        position: relative;

        &::after {
          content: '';
          display: block;
          width: calc(100% - 2px);
          outline: 1px solid var(--white);
          position: absolute;
          bottom: -1px;
          left: 1px;
        }
      }
    }
  }
}
