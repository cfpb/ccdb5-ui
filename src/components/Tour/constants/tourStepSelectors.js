export const DOCUMENT_STEP_SELECTORS = {
  STEP_1: '.content__hero',
  STEP_2: '.m-pagination__btn-next',
  STEP_3: '.saved__search-panel div:nth-child(2)',
  STEP_4: '.back-to-results',
};

export const GEO_STEP_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '.map-shading', label: 'Map Shading' },
  STEP_13: { selector: '#tile-chart-map', label: 'Tile Map' },
  STEP_14: { selector: '.row-chart-section', label: 'Row Charts' },
  STEP_15: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};

export const LIST_COMPLAINTS_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '[data-tour="select-size"]', label: 'Show Number' },
  STEP_13: { selector: '[data-tour="select-sort"]', label: 'Sort Results' },
  STEP_14: {
    selector: '.refine-bar section.m-btn-group',
    label: 'Show Narratives',
  }, // read complaints
  STEP_15: { selector: '.cards-panel', label: 'Results' },
  STEP_16: { selector: '.to-detail', label: 'View Detailed Complaints' }, // complaint id
  STEP_17: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};

export const TRENDS_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '[data-tour="select-lens"]', label: 'Aggregate By' },
  STEP_13: {
    selector: '[data-tour="select-interval"]',
    label: 'Date Interval',
  },
  STEP_14: { selector: '.row-chart-section', label: 'Row Charts' },
  STEP_15: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};
