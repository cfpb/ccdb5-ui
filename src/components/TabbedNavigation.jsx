import './TabbedNavigation.less'
import { connect } from 'react-redux'
import React from 'react'
import { tabChanged } from '../actions/view'

export class TabbedNavigation extends React.Component {
  _setTab( tab ) {
    this.props.onTab( tab );
  }

  _getTabClass( tab ) {
    const tabName = tab.toLowerCase() + ' tab'
    return this.props.tab === tab ? tabName + ' active' : tabName
  }

  render() {
    return (
      <div className="tabbed-navigation">
        <section>
          <button
            className={ this._getTabClass( 'Map' ) }
            onClick={() => this._setTab( 'Map' )}>
            Map
          </button>

          <button
            className={ this._getTabClass( 'Trends' ) }
            onClick={ () => this._setTab( 'Trends' ) }>
            Trends
          </button>

          <button className={ this._getTabClass( 'List' ) }
                  onClick={ () => this._setTab( 'List' ) }>
            List
          </button>
        </section>
      </div>
    );
  }
}

export const mapStateToProps = state => ( {
  tab: state.query.tab
} )

export const mapDispatchToProps = dispatch => ( {
  onTab: tab => {
    dispatch( tabChanged( tab ) )
  }
} )

export default connect( mapStateToProps,
  mapDispatchToProps )( TabbedNavigation )
