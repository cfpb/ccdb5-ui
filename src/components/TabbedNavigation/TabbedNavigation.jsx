import './TabbedNavigation.scss';
import { useDispatch, useSelector } from 'react-redux';
import getIcon from '../Common/Icon/iconMap';
import { selectViewTab } from '../../reducers/view/selectors';
import { tabChanged } from '../../reducers/view/viewSlice';

export const TabbedNavigation = () => {
  const dispatch = useDispatch();
  const tab = useSelector(selectViewTab);

  const getTabClass = (selectedTab) => {
    const tabName = selectedTab.toLowerCase() + ' tab';
    return tab === selectedTab ? tabName + ' active' : tabName;
  };

  return (
    <div className="tabbed-navigation" data-tour="tabbed-navigation">
      <section>
        <button
          className={getTabClass('Trends')}
          onClick={() => dispatch(tabChanged('Trends'))}
        >
          {getIcon('chart')}
          Trends
        </button>

        <button
          className={getTabClass('List')}
          onClick={() => dispatch(tabChanged('List'))}
        >
          {getIcon('list')}
          List
        </button>

        <button
          className={getTabClass('Map')}
          onClick={() => dispatch(tabChanged('Map'))}
        >
          {getIcon('map')}
          Map
        </button>
      </section>
    </div>
  );
};
