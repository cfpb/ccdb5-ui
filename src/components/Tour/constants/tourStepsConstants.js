import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../../../constants';

import {
  GEO_STEP_SELECTORS as GEO,
  LIST_COMPLAINTS_SELECTORS as LIST,
  TRENDS_SELECTORS as TRENDS,
} from './tourStepSelectors';

const generateLinkText = (index, linkObject) =>
  `<li><button class="a-btn a-btn--link" onclick="document.querySelectorAll('.introjs-bullets li a')[${
    index + 1
  }].click()">` +
  linkObject[1].label +
  '</button></li>';

let geoIndex = '';

// we exclude the first entry so we can have Index appear as the title
Object.entries(GEO)
  .slice(1)
  .forEach((value, key) => {
    geoIndex += generateLinkText(key, value);
  });

let listIndex = '';
Object.entries(LIST)
  .slice(1)
  .forEach((value, key) => {
    listIndex += generateLinkText(key, value);
  });

let trendsIndex = '';
Object.entries(TRENDS)
  .slice(1)
  .forEach((value, key) => {
    trendsIndex += generateLinkText(key, value);
  });

export const TOUR_STEPS = {
  [MODE_MAP]: [
    {
      element: GEO.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        geoIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide',
    },
    {
      element: GEO.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: GEO.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available — Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the Map view, which displays complaints by state.' +
        '<br /><br />' +
        'To read individual complaints, select List view' +
        '<br /><br />' +
        'To visualize complaint data with product and issue breakdowns, select Trends view',
      position: 'top',
    },
    {
      element: GEO.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: GEO.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: GEO.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: GEO.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: GEO.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: GEO.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: GEO.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: GEO.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: GEO.STEP_12.selector,
      intro:
        'Map shading defaults to total complaints, but you can change this to show complaints per 1,000 population in that state.',
    },
    {
      element: GEO.STEP_13.selector,
      intro:
        'Hover over a tile on the map to view complaint information specific to that state. You can filter to see data specific to a state by selecting the state or using the filter panel.',
    },
    {
      element: GEO.STEP_14.selector,
      intro:
        'View complaint volume by product in the bar chart. Use the blue arrow by each product to view breakdowns by sub-products.' +
        '<br /><br />Once the product is expanded, select the “Visualize” link as shown below, to view trends on sub-products and issues:',
    },
    {
      element: GEO.STEP_15.selector,
      intro:
        'To learn how to use Trends and List views, select the view and then start the tour.',
    },
  ],
  [MODE_LIST]: [
    {
      element: LIST.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        listIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide',
    },
    {
      element: LIST.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: LIST.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available — Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the List view, which shows individual complaints.' +
        '<br /><br />' +
        'To visualize complaint data with product and issue breakdowns, select Trends view.' +
        '<br /><br />' +
        'To view complaints by state, select Map view.',
    },
    {
      element: LIST.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: LIST.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: LIST.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: LIST.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: LIST.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: LIST.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: LIST.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: LIST.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: LIST.STEP_12.selector,
      intro: 'Select the number of complaints to view per page here.',
    },
    {
      element: LIST.STEP_13.selector,
      intro:
        'Sort complaints in the list view by “Newest to oldest,” “Oldest to newest,” “Relevance,” and “Relevance (asc).”' +
        '<br /><br />' +
        'Relevancy are based on any applied search terms.',
    },
    {
      element: LIST.STEP_14.selector,
      intro:
        'You can see all complaints or focus on those with consumer complaint narratives.' +
        '<br /><br />' +
        'We publish the consumer’s narrative description of what happened from their complaint if the consumer opts to share it publicly and after taking steps to remove personal information.',
    },
    {
      element: LIST.STEP_15.selector,
      intro:
        'These are your search results. As you apply filters and search terms, your results will display here. This view shows a preview of the complaint.',
    },
    {
      element: LIST.STEP_16.selector,
      intro:
        'Click the blue ID number to see more details about a particular complaint.',
    },
    {
      element: LIST.STEP_17.selector,
      intro:
        'To learn how to use Map and Trend views, select the view and then start the tour.',
    },
  ],
  [MODE_TRENDS]: [
    {
      element: TRENDS.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        trendsIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide first-step',
    },
    {
      element: TRENDS.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: TRENDS.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available—Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the Trends view, which visualizes complaint data using a trend line with product and issue breakdowns.' +
        '<br /><br />' +
        'To read individual complaints, select List view.' +
        '<br /><br />' +
        'To view complaints by state, select Map view.',
      position: 'top',
    },
    {
      element: TRENDS.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: TRENDS.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: TRENDS.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: TRENDS.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: TRENDS.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: TRENDS.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: TRENDS.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: TRENDS.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: TRENDS.STEP_12.selector,
      intro:
        'Select the dropdown to aggregate the data by product or by the company to which the CFPB sent the complaint for response.',
    },
    {
      element: TRENDS.STEP_13.selector,
      intro:
        'Select this dropdown to change the time interval displayed on the trend line.',
    },
    {
      element: TRENDS.STEP_14.selector,
      intro:
        'View complaint volume by product in the bar chart. Use the blue arrow by each product to view breakdowns by sub-products.' +
        '<br /><br />Once the product is expanded, select the “Visualize” link as shown below, to view trends on sub-products and issues:',
    },
    {
      element: TRENDS.STEP_15.selector,
      intro:
        'To learn how to use List and Map views, select the view and then start the tour.',
      position: 'top',
    },
  ],
};
