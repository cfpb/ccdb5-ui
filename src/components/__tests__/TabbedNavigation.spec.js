import configureMockStore from 'redux-mock-store';
import { TabbedNavigation } from '../TabbedNavigation';
import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../../constants';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

/**
 *
 * @param {string} tab - The tab
 * @returns {void}
 */
function setupSnapshot(tab) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    query: {
      tab,
    },
  });

  return renderer.create(
    <Provider store={store}>
      <TabbedNavigation />
    </Provider>,
  );
}

describe('component: TabbedNavigation', () => {
  describe('initial state', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot();
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('shows the List tab', () => {
      const target = setupSnapshot(MODE_LIST);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('shows the Map tab', () => {
      const target = setupSnapshot(MODE_MAP);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('shows the Trends tab', () => {
      const target = setupSnapshot(MODE_TRENDS);
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  // TODO: this needs to be reimplemented using modern testing-library
  // https://kentcdodds.com/blog/why-i-never-use-shallow-rendering
  // describe('buttons', () => {
  //   let cb = null
  //   let target = null
  //
  //   beforeEach( () => {
  //     cb = jest.fn()
  //     target = shallow( <TabbedNavigation onTab={ cb } /> )
  //   } )
  //
  //   it( 'tabChanged is called with Map when the button is clicked', () => {
  //     const prev = target.find( '.tabbed-navigation button' )
  //     prev.simulate( 'click' )
  //     expect( cb ).toHaveBeenCalledWith('Map')
  //   } )
  //
  //   it( 'tabChanged is called with Trends when the button is clicked', () => {
  //     const prev = target.find( '.tabbed-navigation button.trends' )
  //     prev.simulate( 'click' )
  //     expect( cb ).toHaveBeenCalledWith('Trends')
  //   } )
  //
  //   it( 'tabChanged is called with List when the button is clicked', () => {
  //     const prev = target.find( '.tabbed-navigation button.list' )
  //     prev.simulate( 'click' )
  //     expect( cb ).toHaveBeenCalledWith('List')
  //   } )
  // })
});
