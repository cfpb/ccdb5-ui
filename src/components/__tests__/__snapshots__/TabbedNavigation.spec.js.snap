// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: TabbedNavigation initial state renders without crashing 1`] = `
<div
  className="tabbed-navigation"
  data-tour="tabbed-navigation"
>
  <section>
    <button
      className="trends tab"
      onClick={[Function]}
    >
      <svg>
        chart.svg
      </svg>
      Trends
    </button>
    <button
      className="list tab"
      onClick={[Function]}
    >
      <svg>
        list.svg
      </svg>
      List
    </button>
    <button
      className="map tab"
      onClick={[Function]}
    >
      <svg>
        map.svg
      </svg>
      Map
    </button>
  </section>
</div>
`;

exports[`component: TabbedNavigation initial state shows the List tab 1`] = `
<div
  className="tabbed-navigation"
  data-tour="tabbed-navigation"
>
  <section>
    <button
      className="trends tab"
      onClick={[Function]}
    >
      <svg>
        chart.svg
      </svg>
      Trends
    </button>
    <button
      className="list tab active"
      onClick={[Function]}
    >
      <svg>
        list.svg
      </svg>
      List
    </button>
    <button
      className="map tab"
      onClick={[Function]}
    >
      <svg>
        map.svg
      </svg>
      Map
    </button>
  </section>
</div>
`;

exports[`component: TabbedNavigation initial state shows the Map tab 1`] = `
<div
  className="tabbed-navigation"
  data-tour="tabbed-navigation"
>
  <section>
    <button
      className="trends tab"
      onClick={[Function]}
    >
      <svg>
        chart.svg
      </svg>
      Trends
    </button>
    <button
      className="list tab"
      onClick={[Function]}
    >
      <svg>
        list.svg
      </svg>
      List
    </button>
    <button
      className="map tab active"
      onClick={[Function]}
    >
      <svg>
        map.svg
      </svg>
      Map
    </button>
  </section>
</div>
`;

exports[`component: TabbedNavigation initial state shows the Trends tab 1`] = `
<div
  className="tabbed-navigation"
  data-tour="tabbed-navigation"
>
  <section>
    <button
      className="trends tab active"
      onClick={[Function]}
    >
      <svg>
        chart.svg
      </svg>
      Trends
    </button>
    <button
      className="list tab"
      onClick={[Function]}
    >
      <svg>
        list.svg
      </svg>
      List
    </button>
    <button
      className="map tab"
      onClick={[Function]}
    >
      <svg>
        map.svg
      </svg>
      Map
    </button>
  </section>
</div>
`;
